dvc/data/db/webdav.py

```diff
@@ -0,0 +1,27 @@
+from typing import TYPE_CHECKING
+
+from dvc.objects.db import ObjectDB
+
+if TYPE_CHECKING:
+    from dvc.fs.base import FileSystem
+
+
+class WebdavObjectDB(ObjectDB):
+    def __init__(self, fs: "FileSystem", path: str, **config):
+        """
+        WebdavFS expects the relative path instead of full prefix.
+
+        So, for example, if:
+            base_url = 'https://server.org/dav/'
+            file url = 'https://server.org/dav/path/to/file.txt'
+
+        WebdavFS expects relative path 'path/to/file.txt' instead of
+        base_path 'dav/path/to/file.txt'. So, we are setting the
+        fs_path to be '' (root) itself instead of 'dav'.
+
+        Note that this should be considered a temporary solution, until
+        we find a better solution for filesystems that don't support full
+        path.
+        """
+        super().__init__(fs, "", **config)
```

Comment:
We use `fs_path` to keep track of prefix, but `WebdavFS` already handles that itself, so what we need is only the subpaths.